train.py

```diff
@@ -118,6 +180,9 @@ def train_epoch(
     total = 0.0
     correct = 0.0
     for i, data in tqdm(enumerate(trainloader, 0)):
+
+        if i > 15 and opt_params['use_devset'] == True:
+            break
         # Indicator sum of gradient less than C for this batch
```

Comment:
Looks like it breaks after the first 15 batches? Is this intended?

Reply:
Yeah, we added this to enable training on a subset of data like we do in the centralized version. We had to do it this way because taking a random sample from the torch dataset wasn't working as expected.